app/questionnaire/placeholder_transforms.py

```diff
@@ -297,8 +299,13 @@ def first_non_empty_item(self, items: Sequence[str]) -> str:
         'items' must be a data structure that preserves order, ie tuple, list etc.
         If order is not important, this can be reused to return `one of` the elements which is non empty.
         """
-        for item in self.remove_empty_from_list(items):
-            return item
+        MIN_LIST_LENGTH = 1
+        filtered_list = self.remove_empty_from_list(items)
+
+        for item in filtered_list:
+            if len(filtered_list) == MIN_LIST_LENGTH:
+                return item
+            return filtered_list[0] if self.block_on_routing_path else filtered_list[1]
 
         return ""
```

Comment:
I don't really understand why the for loop is necessary here, and in what case we return "" below, since the iterator `item` isn't really used inside the for loop?

Reply:
Updated the logic. So it first checks if the ` filtered list` is empty. If it is then it `returns ""`.
I have tested it, but could you also test it locally on runner to see if the solution still works. Thanks 👍